### ISSUE TITLE
[ExpandableTile] height should accommodate large above-the-fold content

### ISSUE THREAD

**Issue by johannesmols** [carbon-design-system/carbon-components-svelte#1730]
I have a simple expandable tile with lots of text in the above section, which is cut off:
![image](https://user-images.githubusercontent.com/11809436/233340236-4a025b00-8d9c-4afd-8a00-06f098bc40c9.png)

When I expand the tile and collapse it again, it shows everything:
![image](https://user-images.githubusercontent.com/11809436/233340286-7c2a5f1f-74b7-49eb-b669-6968ffc6487d.png)

How can I achieve that the height from the second picture is always used even before interacting with it?
I know I might be able to use `tileMaxHeight`, but I don't know the height of the text beforehand, as they are different from text to text.

My tile:
```
<ExpandableTile>
	<div slot="above">
		<p><strong>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porttitor, dolor quis rhoncus venenatis, turpis sapien auctor metus, at porttitor ex quam in justo. Duis eget tortor pretium, gravida sapien in, aliquam leo. Ut imperdiet aliquam dui, non laoreet purus varius vel. Nullam dictum magna leo, at vehicula est bibendum eu.</strong></p>
	</div>
</ExpandableTile>
```

**Comment by well300** [comment]
👏

**Comment by theetrain** [comment]
Hello, it is possible to set a default height to match its contents with `onMount`. Here is a demo: https://svelte.dev/repl/c2d5ddd0bad84c12b2923fa90e6f1828?version=3.59.1

In brief:

```html
<script>
	import { ExpandableTile } from 'carbon-components-svelte@0.74.0/src/Tile'
	import { onMount } from 'svelte'
	
	let tile
	let expanded = true
	let height
	
	const copy = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porttitor, dolor quis rhoncus venenatis, turpis sapien auctor metus, at porttitor ex quam in justo. Duis eget tortor pretium, gravida sapien in, aliquam leo. Ut imperdiet aliquam dui, non laoreet purus varius vel. Nullam dictum magna leo, at vehicula est bibendum eu.'
	
	onMount(() => {
		height = tile.offsetHeight
		expanded = false
	})
</script>

<ExpandableTile bind:ref={tile} tileMaxHeight={height} expanded={expanded}>
  {copy}
</ExpandableTile>
```

Though if you do it this way, it could be a Tile instead of an ExpandableTile since expanding a full-height tile doesn't serve much purpose. Can you describe your use case?

**Comment by johannesmols** [comment]
Hi @theetrain, thank you for your answer. I realized that I didn't properly try to reproduce before posting. I'm actually using this inside of an Accordion item, which I believe is causing the behaviour I posted. I am not trying to set the height to full-size, that would indeed make no sense.

Here an REPL of the issue: https://svelte.dev/repl/ce3cc5d7e5dd410ebcbea81324b1b07a?version=3.59.1

So when you open the accordion the tile is shown like this:
![image](https://github.com/carbon-design-system/carbon-components-svelte/assets/11809436/439972b9-d4ed-49c1-8bb2-94c83bbcef40)

Once you expand the tile and collapse it again, it sets it to the max height that it should have been from the start. I hope that makes sense. Maybe it's unusual to place an expandable tile inside an accordion, but it makes sense for my use case and this seems like a bug.

**Comment by theetrain** [comment]
I tested this with Carbon React and indeed your 'above the fold' content should be clearly visible, and the Tile's height should adjust to accommodate it.  

Working demo: https://stackblitz.com/edit/react-ts-xbmzfq?file=App.tsx

Let me know if you are able to submit a PR.

As for your Accordion example, if you intend to have multiple ExpandableTiles within Accordion content, I recommend to instead have headings and a table of contents instead of Accordions. I checked the [Accordion guidelines](https://carbondesignsystem.com/components/accordion/usage#content) and there's no specific advice here, but my gut tells me to reduce nested interactivity as much as possible.

**Comment by johannesmols** [comment]
Thank you for the info and identifying it as a bug. I'm not sure when and if I have the time to make a PR, as I have a lot going on (and honestly am not great with JS and CSS). I will try to make room for it at some point though.

And yes, I also wasn't too sure about placing expandable elements inside an accordion. My setup is that I have a combo box to select a "category" of information, and then show all instances of this in an accordion, with each instance containing multiple events that are shown as expandable tiles. It might be better to use two combo boxes though to reduce nesting of expandable elements.

**Comment by SimpleProgrammingAU** [comment]
I think I might have found a workaround. The workaround may point us in the direction of a solution, too:
```svelte
<script>
  import { Accordion, AccordionItem, ExpandableTile } from "carbon-components-svelte";

  const copy = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porttitor, dolor quis rhoncus venenatis, turpis sapien auctor metus, at porttitor ex quam in justo. Duis eget tortor pretium, gravida sapien in, aliquam leo. Ut imperdiet aliquam dui, non laoreet purus varius vel. Nullam dictum magna leo, at vehicula est bibendum eu.';
  
  // Set the starting state of the AccordionItem
  let open = false;
</script>

<Accordion>
  <AccordionItem bind:open title="Expanding this doesn't set initial height of tile correctly">
    <ExpandableTile tileMaxHeight={open ? 0 : 1}>
      <div slot="above">
        <p><strong>{copy}</strong></p>
      </div>
      <div slot="below">
        <p>{copy}</p>
      </div>
    </ExpandableTile>
  </AccordionItem>
</Accordion>
``` 

By changing the value of `tileMaxHeight` to zero when the `Accordion` is opened, it triggers `onUpdate` inside `ExpandableTile` that will recalculate the expected height of the tile.

Demo: https://svelte.dev/repl/7292563f37364d3fa14793b40010a770?version=3.59.1

**Comment by metonym** [comment]
As was mentioned above, I agree using a resize observer is the best option, especially since the height of the tile can change dynamically (e.g., resizing the browser width).

See PR #1738

**Comment by theetrain** [comment]
Fix has been released in [v0.75.1](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.75.1)
